services/calculate.go

```diff
@@ -167,152 +100,155 @@ func GetResponseHeaderScore(url string) (totalScore int, XSSReportURL string, ma
-			score = 5
+func GetHSTSScore(HSTS string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if HSTS != "" {
+		if strings.HasPrefix(HSTS, HSTSValues[0]) {
+			totalScore += 4
+			if strings.Contains(HSTS, HSTSValues[1]) || strings.Contains(HSTS, HSTSValues[2]) {
+				totalScore++
+			}
 		}
+	} else {
+		totalScore += 2
 	}
-	return
+	return totalScore, maxScore
+}
+
+// GetCSPScore returns the score for Content Security Policy Header
+func GetCSPScore(CSP string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if CSP != "" {
+		totalScore += 5
+	} else {
+		totalScore += 3
+	}
+	return totalScore, maxScore
+}
+
+// GetPKPScore returns the score for Public Key Pinning Header
+func GetPKPScore(PKP string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if PKP != "" {
+		totalScore += 5
+	} else {
+		totalScore += 3
+	}
+	return totalScore, maxScore
 }
 
 // GetReferrerPolicyScore returns the HTTP Referrer-Policy Response Header Score of the URL
-func GetReferrerPolicyScore(ReferrerPolicy string) (score int) {
-	ReferrerPolicy = strings.TrimSpace(strings.ToLower(ReferrerPolicy))
-	if strings.Compare(ReferrerPolicy, ReferrerPolicyValues[0]) == 0 {
-		score = 5
-	} else if strings.Compare(ReferrerPolicy, ReferrerPolicyValues[1]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[2]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[3]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[4]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[5]) == 0 || strings.Compare(ReferrerPolicy, ReferrerPolicyValues[6]) == 0 {
-		score = 4
-	} else if strings.Compare(ReferrerPolicy, ReferrerPolicyValues[7]) == 0 {
-		score = 2
+func GetReferrerPolicyScore(ReferrerPolicy string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if ReferrerPolicy != "" {
+		ReferrerPolicy = strings.TrimSpace(strings.ToLower(ReferrerPolicy))
+		if score, ok := ReferrerPolicyValues[ReferrerPolicy]; ok {
+			totalScore += score
+		}
 	}
-	return
+	return totalScore, maxScore
 }
 
 // GetXContentTypeScore returns the score for X-Content-Type-Options Header
-func GetXContentTypeScore(XContentType string) (score int) {
-	score = 0
-	if strings.EqualFold(XContentType, XContentTypeHeaderValue) {
-		score = 5
+func GetXContentTypeScore(XContentType string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if XContentType == XContentTypeHeaderValue {
+		totalScore += 5
 	}
-	return
+	return totalScore, maxScore
 }
 
 // GetHTTPVersionScore returns the score for HTTP Version
-func GetHTTPVersionScore(Proto string) (score int) {
-	score = 0
-	if strings.EqualFold(Proto, HTTPVersion[0]) {
-		score = 5
-	} else if strings.EqualFold(Proto, HTTPVersion[1]) {
-		score = 2
-	}
-	return
+func GetHTTPVersionScore(Proto string, totalScore int, maxScore int) (int, int) {
+	maxScore += 5
+	if Proto == HTTPVersion[0] {
```

Comment:
@maruthi-adithya does this method handle it when the protocol case is different? string.equalFold ignores casing between strings, but == doesnt, so we should probably normalize all strings, maybe to lowercase or uppercase.